src/index.js

```diff
@@ -90,90 +81,109 @@ class PgBoss extends EventEmitter {
   }
 
   async start () {
-    if (!this.stopped) {
-      return this
+    if (this.starting || this.started) {
+      return
     }
 
+    this.starting = true
+
     if (this.db.isOurs && !this.db.opened) {
       await this.db.open()
     }
 
-    await this.contractor.start()
-
-    this.stopped = false
-    this.started = true
+    if (this.config.migrate) {
+      await this.contractor.start()
+    } else {
+      await this.contractor.check()
+    }
 
     this.manager.start()
 
-    if (!this.config.noSupervisor) {
+    if (this.config.supervise) {
       await this.boss.supervise()
     }
 
-    if (!this.config.noScheduling) {
+    if (this.config.monitorStateIntervalSeconds) {
+      await this.boss.monitor()
+    }
+
+    if (this.config.schedule) {
       await this.timekeeper.start()
     }
 
+    this.starting = false
+    this.started = true
+    this.stopped = false
+
     return this
   }
 
   async stop (options = {}) {
-    if (this.stoppingOn) {
+    if (this.stoppingOn || this.stopped) {
       return
     }
 
-    if (this.stopped) {
-      this.emit(events.stopped)
-    }
-
-    let { destroy = false, graceful = true, timeout = 30000 } = options
+    let { destroy = false, graceful = true, timeout = 30000, wait = false } = options
```

Comment:
Can we make `wait=true` the default behavior in v10? I think nobody wants to forcefully exit jobs by default.